File: Grind 75 in Java/0150-evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        LinkedList<Integer> stack = new LinkedList<>();
        HashSet<String> ops = new HashSet<>(Set.of("+", "-", "*", "/"));

        for (String t : tokens) {
            if (ops.contains(t)) {
                int second = stack.removeLast();
                int first = stack.removeLast();
                int sum = 0;

                switch (t) {
                    case "+":
                        sum = first + second;
                        break;
                    case "-":
                        sum = first - second;
                        break;
                    case "*":
                        sum = first * second;
                        break;
                    case "/":
                        sum = first / second;
                        break;
                }

                stack.addLast(sum);
            } else {
                stack.addLast(Integer.parseInt(t));
            }
        }

        return stack.removeLast();
    }
}

File: leetcode in kotlin/0150-evaluate-reverse-polish-notation.kt
class Solution {
    fun evalRPN(tokens: Array<String>) = with (LinkedList<Int>()) {
        val ops = listOf("+", "-", "*", "/")

        for (token in tokens) {
            if (token in ops) {
                val b = removeLast()
                val a = removeLast()

                when (token) {
                    "+" -> addLast(a + b)
                    "-" -> addLast(a - b)
                    "*" -> addLast(a * b)
                    "/" -> addLast(a / b)
                }
            } else {
                addLast(token.toInt())
            }
        }

        removeLast()
    }
}
